x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1581,6 +1592,16 @@ public List<AggregationSpec> getAggregations() {
         );
     }
 
+    @Override
+    public Map<String, Mapper.TypeParser> getMappers() {
```

Comment:
What happens if the ML plugin is disabled, say because there's some problem with the native code? I guess the mapper is still available to use, because this method doesn't check the value of `this.enabled`?

The scenario I am thinking about is:

1. User uses this new mapper
2. Following an upgrade there's some fatal problem with the ML native code on their machines, so they are advised to disable the ML plugin so that they can at least get their cluster running with non-ML functionality
3. After disabling the ML plugin, their indices that use the new mapper are fatally broken

I am hoping that 3 wouldn't happen because the mapper _would_ still be available.

Reply:
@droberts195 I am sticking in ML just as a place to be. We can move it around where ever.